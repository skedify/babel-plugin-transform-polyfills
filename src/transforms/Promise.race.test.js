describe('Promise.race', () => {
  it('should replace Promise.race with a polyfill', () => {
    const result = compile(
      `
      const c = Promise.race([])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Promise.race([Promise.race([1, 2])])
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Promise.race([]))
    `
    )

    expect(result).toMatchSnapshot()
  })
})
