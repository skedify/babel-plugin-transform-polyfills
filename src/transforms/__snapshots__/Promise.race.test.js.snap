// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise.race should not import polyfill more than once 1`] = `
"import _Promise_race from \\"babel-plugin-transform-polyfills/polyfills/Promise/race\\";

const e = _Promise_race([_Promise_race([1, 2])]);"
`;

exports[`Promise.race should replace Promise.race with a polyfill 1`] = `
"import _Promise_race from \\"babel-plugin-transform-polyfills/polyfills/Promise/race\\";

const c = _Promise_race([]);"
`;

exports[`Promise.race should work when part of a call chain 1`] = `
"import _Promise_race from \\"babel-plugin-transform-polyfills/polyfills/Promise/race\\";

const c = Object.keys(_Promise_race([]));"
`;
